### PR TITLE
(PWGCF): Analysis moved to JCorran

### PR DESCRIPTION
**Pull request by MaximVirta**
Suggestions by @victor-gonzalez applied and analysis moved to JCorran folder

### CODE REVIEW

File: PWGCF/Flow/Tasks/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
 
+
 o2physics_add_dpl_workflow(flow-pt-efficiency
                     SOURCES flowPtEfficiency.cxx
                     PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore
```
Comment (victor-gonzalez) [inline PWGCF/Flow/Tasks/CMakeLists.txt:15]: Why is this file being modified?
Are the proposed changes rebased to the latest version?

File: PWGCF/JCorran/Core/FlowJSPCAnalysis.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+/* Header files. */
+#include <iostream>
+#include <array>
+#include <vector>
+#include <TComplex.h>
+#include <TProfile.h>
+
+// O2 headers. //
+// #include "Framework/AnalysisDataModel.h"
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Core/FlowJSPCAnalysis.h:26]: Please, don't leave code commented

File: PWGCF/JCorran/Core/FlowJSPCAnalysis.h [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+#include "Framework/HistogramRegistry.h"
+#include "PWGCF/JCorran/Core/JQVectors.h"
+// #include "Framework/StaticFor.h"
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Core/FlowJSPCAnalysis.h:29]: Same here

File: PWGCF/JCorran/Core/FlowJSPCObservables.h [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+// or submit itself to any jurisdiction.
+
+// \brief   Helper class for the SPC-related analyses.
+// \author  Maxim Virta (maxim.virta@cern.fi)
+
+#ifndef PWGCF_JCORRAN_CORE_FLOWJSPCOBSERVABLES_H_
+#define PWGCF_JCORRAN_CORE_FLOWJSPCOBSERVABLES_H_
+
+// O2 headers. //
+// #include "Framework/AnalysisDataModel.h"
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Core/FlowJSPCObservables.h:19]: Same here

File: PWGCF/JCorran/Core/FlowJSPCObservables.h [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+#include "Framework/HistogramRegistry.h"
+// #include "Framework/StaticFor.h"
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Core/FlowJSPCObservables.h:21]: And here

File: PWGCF/JCorran/Tasks/CMakeLists.txt [hunk 6]
```diff
@@ -13,3 +13,13 @@ o2physics_add_dpl_workflow(jfluc-analysis
                     SOURCES jflucAnalysisTask.cxx
                     PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore O2Physics::JCorran
                     COMPONENT_NAME Analysis)
+
+o2physics_add_dpl_workflow(flow-spc-task
+                    SOURCES flowJSPCAnalysisTask.cxx
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Tasks/CMakeLists.txt:18]: Please, do not use the name task neither in the executable not in the source filename

File: PWGCF/JCorran/Tasks/CMakeLists.txt [hunk 7]
```diff
@@ -13,3 +13,13 @@ o2physics_add_dpl_workflow(jfluc-analysis
...
+                    PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::AnalysisCore O2Physics::JCorran
+                    COMPONENT_NAME Analysis)
+
+o2physics_add_dpl_workflow(flow-nuacreation-task
+                    SOURCES flowJNUACreationTask.cxx
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Tasks/CMakeLists.txt:23]: Same here

File: PWGCF/JCorran/Core/FlowJHistManager.h [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+#include <string_view>
+#include "TH1.h"
+#include "TH2.h"
+#include "TH3.h"
+#include "TProfile.h"
+#include "TProfile2D.h"
+
+// O2 headers. //
+#include "Framework/HistogramRegistry.h"
+// #include "Framework/StaticFor.h"
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Core/FlowJHistManager.h:32]: Please, don't leave code commented

**Review by victor-gonzalez — CHANGES_REQUESTED** [review]
Please, consider my suggestions, specially the ones referring to files which should not be modified and the one concerning to the naming of executables and source files

File: PWGCF/Flow/Tasks/CMakeLists.txt (hunk 1 above)
Comment (MaximVirta) [reply PWGCF/Flow/Tasks/CMakeLists.txt:15]: I moved the Task files to JCorran, they were here before but an empty space got left behind apparently. I'll remove it.

File: PWGCF/JCorran/Core/FlowJHistManager.h (hunk 8 above)
Comment (MaximVirta) [reply PWGCF/JCorran/Core/FlowJHistManager.h]: Done

File: PWGCF/JCorran/Core/FlowJSPCAnalysis.h (hunk 3 above)
Comment (MaximVirta) [reply PWGCF/JCorran/Core/FlowJSPCAnalysis.h]: Done

File: PWGCF/JCorran/Core/FlowJSPCObservables.h (hunk 4 above)
Comment (MaximVirta) [reply PWGCF/JCorran/Core/FlowJSPCObservables.h]: Done

File: PWGCF/JCorran/Core/FlowJSPCObservables.h (hunk 5 above)
Comment (MaximVirta) [reply PWGCF/JCorran/Core/FlowJSPCObservables.h]: Done

File: PWGCF/JCorran/Tasks/CMakeLists.txt (hunk 6 above)
Comment (victor-gonzalez) [reply PWGCF/JCorran/Tasks/CMakeLists.txt]: Well, the task files are in the `Tasks` directory and the class files should go into the `Core` directory

Comment (MaximVirta) [reply PWGCF/JCorran/Tasks/CMakeLists.txt]: Alright, I renamed them now.

File: PWGCF/JCorran/Tasks/flowJNUACreation.cxx [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+using namespace std;
+
+using MyCollisions = soa::Join<aod::Collisions, aod::EvSels, aod::Mults,
+                               aod::FT0sCorrected, aod::CentFT0Ms,
+                               aod::CentFT0As, aod::CentFT0Cs, aod::CentFV0As,
+                               aod::CentFDDMs, aod::CentNTPVs>;
+
+using MyTracks = soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA>;
+
+struct flowJNUACreation {
```
Comment (victor-gonzalez) [inline PWGCF/JCorran/Tasks/flowJNUACreation.cxx:54]: In here you are free to use `Task` as suffix if so you want
I did not complain about it before

**Review by victor-gonzalez — APPROVED** [review]
Thanks for addressing my comments!
For the next iteration consider what I suggested in my latest comments if so you want

**Pull request merged**